perf/src/sigverify.rs

```diff
@@ -309,17 +238,89 @@ fn do_get_packet_offsets(
     ))
 }
 
-fn get_packet_offsets(packet: &Packet, current_offset: usize) -> PacketOffsets {
+fn get_packet_offsets(packet: &mut Packet, current_offset: usize) -> PacketOffsets {
     let unsanitized_packet_offsets = do_get_packet_offsets(packet, current_offset);
     if let Ok(offsets) = unsanitized_packet_offsets {
+        if let Some(msg_start_offset) = (offsets.msg_start as usize).checked_sub(current_offset) {
+            check_for_simple_vote_transaction(packet, msg_start_offset).ok();
+        }
         offsets
     } else {
         // force sigverify to fail by returning zeros
         PacketOffsets::new(0, 0, 0, 0)
     }
 }
 
-pub fn generate_offsets(batches: &[Packets], recycler: &Recycler<TxOffset>) -> TxOffsets {
+fn check_for_simple_vote_transaction(
+    packet: &mut Packet,
+    msg_start_offset: usize,
+) -> Result<(), PacketError> {
+    let message_account_keys_len_offset = msg_start_offset
+        .checked_add(MESSAGE_HEADER_LENGTH)
+        .ok_or(PacketError::InvalidLen)?;
+
+    let (pubkey_len, pubkey_len_size) =
+        decode_shortu16_len(&packet.data[message_account_keys_len_offset..])
+            .map_err(|_| PacketError::InvalidLen)?;
+
+    let pubkey_start = message_account_keys_len_offset
+        .checked_add(pubkey_len_size)
+        .ok_or(PacketError::InvalidLen)?;
+
+    let instructions_len_offset = pubkey_len
+        .checked_mul(size_of::<Pubkey>())
+        .and_then(|v| v.checked_add(pubkey_start))
+        .and_then(|v| v.checked_add(size_of::<Hash>()))
+        .ok_or(PacketError::InvalidLen)?;
+
+    // Packet should have 1 more byte for instructions.len
```

Comment:
shortu16 can read up to 3 bytes (look up `MAX_ENCODING_LENGTH`)

Reply:
hmm but could there only be 1 byte remaining if the length is 0. Maybe we skip the `decode_shortu16_len` call and read the next byte directly?

Reply:
per our chat, will add a function does something like this: is_length_eq(&Packet, offset, value) -> bool

Reply:
Sorry, disregard this. If you look at `decode_shortu16_len`, it will error if there aren't enough bytes to read from, so this is safe.

Reply:
> Sorry, disregard this. If you look at `decode_shortu16_len`, it will error if there aren't enough bytes to read from, so this is safe.

lovely